src/jllvm/vm/JIT.hpp

```diff
@@ -41,6 +43,18 @@
 
 namespace jllvm
 {
+
+/// Where code should be executed by default.
+enum class ExecutionMode
+{
+    /// Executes code in the JIT whenever possible.
+    JIT,
+    /// Executes code in the interpreter whenever possible.
+    Interpreter,
+    /// Dynamically adjusts where code is being executed.
+    Mixed
+};
+
```

Comment:
Would this not belong in VM?

Reply:
I would prefer that as well, but this would create a circular dependency between `VirtualMachine.hpp` and `JIT.hpp` which cannot really be easily broken up. 
Given that the JIT is the actual consumer of this enum I think its okay for now, but probably worth revisiting in the future.